test/integration/singlecluster/scheduler/fairsharing/fair_sharing_test.go

```diff
@@ -508,6 +508,154 @@ var _ = ginkgo.Describe("Scheduler", func() {
+
+			ginkgo.By("Verifying one workload from each lq is admitted")
+			util.ExpectWorkloadsToBeAdmittedCount(ctx, k8sClient, 1, lqAWls...)
+			util.ExpectWorkloadsToBeAdmittedCount(ctx, k8sClient, 1, lqBWls...)
+		})
+
+		ginkgo.It("prioritizes workloads from less active LocalQueues to maintain fairness", func() {
+			ginkgo.By("Saturating the cq with lq-a")
+			initialWls := []*kueue.Workload{
+				createWorkload("lq-a", "4"),
+				createWorkload("lq-a", "4"),
+			}
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, initialWls...)
+
+			ginkgo.By("Creating pending workloads for lq-a")
+			lqAWls := []*kueue.Workload{
+				createWorkload("lq-a", "4"),
+				createWorkload("lq-a", "4"),
+				createWorkload("lq-a", "4"),
+			}
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, lqAWls...)
+
+			ginkgo.By("Creating a pending workload for lq-b")
+			wlB := createWorkload("lq-b", "4")
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, wlB)
+
+			ginkgo.By("Releasing quota")
+			util.FinishWorkloads(ctx, k8sClient, initialWls...)
+
+			ginkgo.By("Verifying workload from lq-b is admitted")
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wlB)
+		})
+
+		ginkgo.It("admits workload from new LocalQueue when all others have high usage", func() {
+			ginkgo.By("Saturating the cq with lq-a and lq-b")
+			initialWls := []*kueue.Workload{
+				createWorkload("lq-a", "4"),
+				createWorkload("lq-b", "4"),
+			}
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, initialWls...)
+
+			ginkgo.By("Creating pending workloads for lq-a and lq-b")
+			lqAWls := []*kueue.Workload{
+				createWorkload("lq-a", "4"),
+				createWorkload("lq-a", "4"),
+			}
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, lqAWls...)
+			lqBWls := []*kueue.Workload{
+				createWorkload("lq-b", "4"),
+				createWorkload("lq-b", "4"),
+			}
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, lqBWls...)
+
+			ginkgo.By("Creating a pending workload for lq-c")
+			wlC := createWorkload("lq-c", "4")
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, wlC)
+
+			ginkgo.By("Releasing quota")
+			util.FinishWorkloads(ctx, k8sClient, initialWls...)
+
+			ginkgo.By("Verifying workload from lq-c is admitted")
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wlC)
+		})
+
+		ginkgo.It("admits workloads from less active LocalQueues after quota is released", func() {
+			ginkgo.By("Saturating the cq with lq-a")
+			initialWls := []*kueue.Workload{
+				createWorkload("lq-a", "4"),
+				createWorkload("lq-a", "4"),
+			}
+			util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, initialWls...)
+
+			ginkgo.By("Creating pending workloads for lq-b")
+			lqBWls := []*kueue.Workload{
+				createWorkload("lq-b", "4"),
+				createWorkload("lq-b", "4"),
+			}
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, lqBWls...)
+
+			ginkgo.By("Creating a pending workload for lq-c")
+			wlC := createWorkload("lq-c", "4")
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, wlC)
+
```

Comment:
Let's make sure at this point that the usage for LQ-A > 0 and LQ-B and LQ-C is equal to 0